fossa.c

```diff
@@ -2484,6 +2484,227 @@ void ns_printf_http_chunk(struct ns_connection *nc, const char *fmt, ...) {
   /* LCOV_EXCL_STOP */
 }
 
+int ns_http_parse_header(struct ns_str *hdr, const char *var_name,
+                         char *buf, size_t buf_size) {
+  int ch = ' ', ch1 = ',', len = 0, n = strlen(var_name);
+  const char *p, *end = hdr->p + hdr->len, *s = NULL;
+
+  if (buf != NULL && buf_size > 0) buf[0] = '\0';
+
+  /* Find where variable starts */
+  for (s = hdr->p; s != NULL && s + n < end; s++) {
+    if ((s == hdr->p || s[-1] == ch || s[-1] == ch1) && s[n] == '=' &&
+        !memcmp(s, var_name, n)) break;
+  }
+
+  if (s != NULL && &s[n + 1] < end) {
+    s += n + 1;
+    if (*s == '"' || *s == '\'') {
+      ch = ch1 = *s++;
+    }
+    p = s;
+    while (p < end && p[0] != ch && p[0] != ch1 && len < (int) buf_size) {
+      if (ch != ' ' && p[0] == '\\' && p[1] == ch) p++;
+      buf[len++] = *p++;
+    }
+    if (len >= (int) buf_size || (ch != ' ' && *p != ch)) {
+      len = 0;
+    } else {
+      if (len > 0 && s[len - 1] == ',') len--;
+      if (len > 0 && s[len - 1] == ';') len--;
+      buf[len] = '\0';
+    }
+  }
+
+  return len;
+}
+
+#ifndef NS_DISABLE_HTTP_DIGEST_AUTH
+static FILE *open_auth_file(const char *path, int is_directory,
+                            const struct ns_serve_http_opts *opts) {
+  char buf[MAX_PATH_SIZE];
+  const char *p;
+  FILE *fp = NULL;
+
+  if (opts->global_auth_file != NULL) {
+    fp = fopen(opts->global_auth_file, "r");
+  } else if (is_directory && opts->per_directory_auth_file) {
+    snprintf(buf, sizeof(buf), "%s%c%s", path, '/',
```

Comment:
I guess the idea here is to use a platform specific DIRSEP. Perhaps better do ing it now or at least adding a TODO marker.